backend/src/main/java/com/example/backend/use_case/route_details/RouteDetailsImplementation.java

```diff
@@ -0,0 +1,42 @@
+package com.example.backend.use_case.route_details;
+
+import com.example.backend.data_access.route.RouteDataAccessInterface;
+import com.example.backend.entity.Route;
+import com.example.backend.entity.RouteDirection;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class RouteDetailsImplementation implements RouteDetailsService {
+    @Autowired
+    private RouteDataAccessInterface routeDAO;
+
+    /** Returns the route details for the given routeTag and dirTag. Returns null if the routeTag or dirTag is invalid.
+     *
+     * @param routeTag String representing the routeTag
+     * @param dirTag String representing the dirTag
+     * @return RouteDetailsOutputData object containing the routeTag, dirTag, dirName, and routeShape. Null if the routeTag or dirTag is invalid.
+     */
+    public RouteDetailsOutputData execute(String routeTag, String dirTag) {
+        Route route = routeDAO.getRouteByRouteTag(routeTag);
+
+        // TODO: Fix error handling
```

Comment:
is this "fixed" yet?

Reply:
In this context, "fix error handling" should be replaced with "implement error handling." The way errors are handled are by returning null, which while it works, isn't a great way to handle these errors. The problem is that error handling in spring boot isn't as simple as just adding a try-catch block or throwing an Exception, there needs to be a not insignificant amount of code written to account for the different errors, for each use case. I've deliberately not implemented this yet because my plan is to go back for each use case and implement error handling later. Currently, the code works on the assumption that the route tag and direction tag are correct, which for our current purposes, is good enough.